src/preprocess_data.py

```diff
@@ -48,10 +48,40 @@ def preprocess_data():
 
     raw_csv_files = glob.glob(f"{input_dir}/*.csv")
 
-    characters_data = pd.concat([pd.read_csv(file) for file in raw_csv_files],
-        ignore_index=True)
+    publisher_dfs = []
+    for csv_file in raw_csv_files:
+        if verbose: print(f"Processing {csv_file}")
+        publisher = os.path.basename(csv_file).split("-")[0]
+        publisher_df = pd.read_csv(csv_file)
+        publisher_df.columns = publisher_df.columns.str.lower().str.replace(' ', '_')
+        publisher_df = publisher_df.drop(["page_id",
+                                          "urlslug",
+                                          "id",
+                                          "alive"], axis=1)
+        publisher_df['publisher'] = publisher
+        publisher_dfs.append(publisher_df)
+
+    characters_data = pd.concat(publisher_dfs, ignore_index=True)
+
+    # There is some weird comic book multiverse stuff going on with this... We
+    # can't remove the bracket aliases without creating duplicate entries
+    #characters_data['name'] = characters_data['name'].str.split('(').str[0]
+    characters_data['align'] = (characters_data['align'].str.split(' ').str[0]).astype("category")
+    characters_data['eye'] = (characters_data['eye'].str.split(' ').str[0]).astype("category")
+    characters_data['hair'] = (characters_data['hair'].str.split(' ').str[0]).astype("category")
+    characters_data['sex'] = (characters_data['sex'].str.split(' ').str[0]).astype("category")
+    characters_data['gsm'] = (characters_data['gsm'].str.split(' ').str[0]).astype("category")
```

Comment:
I have seen you are getting the second words out, that is a good way since all rows has the same. Do we need to take those as factor afterwards?

Reply:
I wanted to collapse the values for those into categoricals (i.e eyes brown, blue ...etc) because I think that's how we'll be predicting. We can always edit these later if we find out we don't want them that way.